visitor.php

```diff
@@ -499,18 +498,33 @@ private function addTags(string $name, Doc $docComment): ?Doc
      */
     private function discoverInheritedArgs(DocBlock $docblock, array $additions): array
     {
-        foreach ($additions as $add) {
-            if ($add->tag === '@phpstan-param') {
-                // We can't yet handle merging args.
-                return $additions;
-            }
-        }
-
         /** @var Param[] $params */
         $params = $docblock->getTagsByName('param');
 
         foreach ($params as $param) {
-            $additions = array_merge($additions, $this->getInheritedTagsForParam($param));
+            $inherited = $this->getInheritedTagsForParam($param);
+
+            if (count($inherited) === 0) {
+                continue;
+            }
+
+            foreach ($additions as $addition) {
+                if ($addition->tag !== '@phpstan-param') {
+                    continue;
+                }
+
+                foreach ($inherited as $inherit) {
+                    if ($addition->name !== $inherit->name) {
+                        continue;
+                    }
+
+                    $addition->children = array_merge($addition->children, $inherit->children);
```

Comment:
`$addition` is never used again.
Does this make a change in `$additionS`? I think not, not without pass-by-reference.

Reply:
```php
<?php

$original = [ 1, 2, 3 ];
foreach ($original as $item) {
    // Change item
    $item = 0;
}
var_dump($original);
```

Reply:
In case of object it works! 😲 

```php
<?php

class Hype { public $value = 1; }

$original = [ new Hype(), new Hype(), new Hype() ];
foreach ($original as $item) {
    // Change item
    $item->value = 0;
}
var_dump($original);
```

Reply:
Yep objects are always passed by reference

Reply:
Could you please uninstall PHP from your computer? :)